tools/test-backend

```diff
@@ -204,4 +246,4 @@ if __name__ == "__main__":
         print('FAILED!')
     else:
         print('DONE!')
-    sys.exit(bool(failures))
+    sys.exit(0)
```

Comment:
I see, well we shouldn't be using subprocess to call `test-backend` recursively anyway, and presumably that error was because there were actual failures.  So I think that'll be fixed automatically when you address the other comments.